crates/module-system/module-implementations/sov-evm/src/call.rs

```diff
@@ -123,6 +124,36 @@ where
         self.pending_transactions
             .push(&pending_transaction, state)?;
 
+        // Fetch `head` and `pending_len` before the `native` code block.
+        // This ensures consistent gas charges between native and non-native execution.
+        #[allow(unused_variables)]
+        let head = self
+            .head
+            .get(state)?
+            .expect("Impossible happened: Head must be set.");
+
+        #[allow(unused_variables)]
+        let pending_len = self.pending_transactions.len(state)?;
```

Comment:
If it's only used in native - why not move it down into the block?

Reply:
Will address in folloup